src/perl/lib/wtsi_clarity/epp/isc/analyte_pooler.pm

```diff
@@ -94,16 +94,18 @@ sub _uniq_array {
   my ($self, @array) = @_;
 
   my %seen;
-  return grep { !$seen{$_}++ } @array;
+  my @uniq_array =  grep { !$seen{$_}++ } @array;
+
+  return \@uniq_array;
 }
 
 sub _get_values_from_nodelist {
   my ($self, $function, $nodelist) = @_;
-  my @values = $self->_uniq_array(
+  my $values = $self->_uniq_array(
```

Comment:
I think there's already a `uniq` function in the `List::MoreUtils` package that we're already using